File: go/hackerrank/archive/20.08/ransomNote.go
// https://www.hackerrank.com/challenges/ctci-ransom-note/problem

func checkMagazine(magazine []string, note []string) {
    m := make(map[string]int)

    for _, word := range(magazine){
        m[word] += 1
    }

    for _, word := range(note){
        if exist := m[word]; exist == 0 {
            fmt.Print("No")
            return
        }

        m[word] -= 1
        if m[word] < 0 {
            fmt.Print("No")
            return
        }
    }

    fmt.Print("Yes")
    return
}

File: python/hackerrank/archive/20.08/ransom_note.py
# https://www.hackerrank.com/challenges/ctci-ransom-note/problem

def checkMagazine(magazine, note):
    count_word = {}
    for word in magazine:
        count_word[word] = 1 if word not in count_word else count_word[word] + 1

    for word in note:
        if word not in count_word:
            print("No")
            return

        count_word[word] -= 1
        if count_word[word] < 0:
            print("No")
            return

    print("Yes")
    return
